### PR TITLE
Feat(authentication): Load auth providers

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Add loader to register providers when module is started
- Create database providers if they don't already exist

### CODE REVIEW

File: packages/authentication/src/loaders/providers.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const loadedProviders = new Map(providers.map((p) => [p.provider, p]))
+
+  const providersToCreate: ServiceTypes.CreateAuthProviderDTO[] = []
+
+  for (const provider of providersToLoad) {
+    container.registerAdd("providers", asClass(provider).singleton())
+
+    container.register({
+      [`provider_${provider.PROVIDER}`]: asClass(provider).singleton(),
+    })
```
Comment (olivermrbl) [inline packages/authentication/src/loaders/providers.ts:36]: question: Would it make sense to register them as `auth_providers_[provider]` and `auth_provider_[provider]`?

File: packages/authentication/src/loaders/providers.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  | ModulesSdkTypes.ModuleServiceInitializeOptions
+  | ModulesSdkTypes.ModuleServiceInitializeCustomDataLayerOptions
+>): Promise<void> => {
+  // if(options.providers?.length) {
+  // TODO: implement plugin provider registration
+  // }
+
+  const providersToLoad = Object.values(defaultProviders)
+
+  const authProviderService: AuthProviderService =
```
Comment (adrien2p) [inline packages/authentication/src/loaders/providers.ts:20]: question: Is there a reason why we don't resolve here?

File: packages/authentication/src/loaders/providers.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const providers = await authProviderService.list({
+    provider: providersToLoad.map((p) => p.PROVIDER),
+  })
+
+  const loadedProviders = new Map(providers.map((p) => [p.provider, p]))
+
+  const providersToCreate: ServiceTypes.CreateAuthProviderDTO[] = []
+
+  for (const provider of providersToLoad) {
+    container.registerAdd("providers", asClass(provider).singleton())
```
Comment (adrien2p) [inline packages/authentication/src/loaders/providers.ts:32]: suggestion: could we not use registerAdd and instead create a providers array that we register after the loop?

**Review by adrien2p — APPROVED** [review]
LGTM, few suggestions :D

**Pull request merged**